File: NMathService/Contracts/Research/ResearchResult.cs
using NMathService.Common;

namespace NMathService.Contracts.Research;

public record ResearchResult(
    string Function,
    string Derivative,
    string SecondOrderDerivative,
    string Parity,
    List<double> UnacceptableArgumentValues,
    double Min,
    double Max,
    List<double> Asymptotes,
    List<Point> Points);

File: nmath-mobile-client/src/types/ResearchResult.ts
import DataPoint from "./DatePoint"

export type ResearchResult = {
  function: string
  derivative: string
  secondOrderDerivative: string
  parity: string
  unacceptableArgumentValues: number[]
  min: number
  max: number
  asymptotes: number[]
  points: DataPoint[]
}
